File: WebApp/WebApp/Models/CatalogueHistory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace WebApp.Models
{
    public class CatalogueHistory
    {
        public int Id { get; set; }
        [Required]
        [Range(1, 15000)]
        public float TicketPrice { get; set; }
        
        [Required]
        public int CatalogueID { get; set; }
        public Catalogue Catalogue { get; set; }
        
        [Required]
        public int TicketTypeID { get; set; }
        public TicketType TicketType { get; set; }
        public List<Ticket> Tickets { get; set; }

    }
}

File: FronEnd/src/app/models/CatalogueHistory.ts
import { Catalogue } from './Catalogue';
import { TicketType } from './TicketType';
import { Ticket } from './Ticket';

export class CatalogueHistory{
    Id: number;
    TicketPrice: number;
    CatalogueID:number;
    TicketTypeID:number;
    Catalogue: Catalogue;
    TicketType: TicketType;
    Tickets: Ticket[];
}